lib/matplotlib/figure.py

```diff
@@ -2789,14 +2862,11 @@ def draw(self, renderer):
             return
 
         artists = self._get_draw_artists(renderer)
-
         try:
             renderer.open_group('figure', gid=self.get_gid())
-            if self.get_constrained_layout() and self.axes:
-                self.execute_constrained_layout(renderer)
-            if self.get_tight_layout() and self.axes:
+            if self.axes and self.get_layout_engine() is not None:
                 try:
-                    self.tight_layout(**self._tight_parameters)
+                    self.get_layout_engine().execute(self)
```

Comment:
Do we want to put a `self.draw_without_render()` call here and then remove that logic from both of the layout engines?  That way all of the implementations can assume that we have done one draw, know the size of everything and have a renderer?

Reply:
... I guess?  CL actually does two draws, so the engines are still likely to do their own draws...

Reply:
I think we can defer this to a later date.